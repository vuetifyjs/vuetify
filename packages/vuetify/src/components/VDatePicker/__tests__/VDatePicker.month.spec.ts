// @ts-nocheck
/* eslint-disable */

// import VDatePicker from '../VDatePicker'
// import { Lang } from '../../../services/lang'
import {
  mount,
  MountOptions,
  Wrapper,
} from '@vue/test-utils'
// import Vue from 'vue'
// import { preset } from '../../../presets/default'
// import en from '../../../locale/en'

// Vue.prototype.$vuetify = {
//   icons: {
//     values: {
//       next: 'mdi-chevron-right',
//       prev: 'mdi-chevron-left',
//     },
//   },
// }

describe.skip('VDatePicker.ts', () => {
  type Instance = InstanceType<typeof VDatePicker>
  let mountFunction: (options?: MountOptions<Instance>) => Wrapper<Instance>
  beforeEach(() => {
    mountFunction = (options?: MountOptions<Instance>) => {
      return mount(VDatePicker, {
        ...options,
        mocks: {
          $vuetify: {
            rtl: false,
            lang: new Lang({
              ...preset,
            }),
          },
        },
      })
    }
  })

  it('should emit input event on year click (reactive picker)', async () => {
    const wrapper = mountFunction({
      propsData: {
        value: '2013-05',
        type: 'month',
        reactive: true,
      },
    })

    wrapper.setData({
      internalActivePicker: 'YEAR',
    })

    const input = jest.fn()
    wrapper.vm.$on('input', input)

    const change = jest.fn()
    wrapper.vm.$on('change', input)

    wrapper.findAll('.v-date-picker-years li.active + li').at(0).trigger('click')
    expect(input).toHaveBeenCalledWith('2012-05')
    expect(change).not.toHaveBeenCalled()
  })

  it('should render flat picker', () => {
    const wrapper = mountFunction({
      propsData: {
        value: '2013-05',
        flat: true,
        type: 'month',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render picker with elevation', () => {
    const wrapper = mountFunction({
      propsData: {
        value: '2013-05',
        elevation: 15,
        type: 'month',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should not emit input event on year click if month is not allowed', async () => {
    const cb = jest.fn()
    const wrapper = mountFunction({
      propsData: {
        value: '2013-05',
        type: 'month',
        allowedDates: () => false,
      },
    })

    wrapper.setData({
      internalActivePicker: 'YEAR',
    })

    wrapper.vm.$on('input', cb)
    wrapper.findAll('.v-date-picker-years li.active + li').at(0).trigger('click')
    expect(cb).not.toHaveBeenCalled()
  })

  it('should emit input event on month click', async () => {
    const cb = jest.fn()
    const wrapper = mountFunction({
      propsData: {
        value: '2013-05',
        type: 'month',
      },
    })

    wrapper.vm.$on('input', cb)
    wrapper.findAll('.v-date-picker-table--month button').at(0).trigger('click')
    expect(cb).toHaveBeenCalledWith('2013-01')
  })

  it('should be scrollable', async () => {
    const wrapper = mountFunction({
      propsData: {
        value: '2013-05',
        type: 'month',
        scrollable: true,
      },
    })

    wrapper.findAll('.v-date-picker-table--month').at(0).trigger('wheel', { deltaY: 1 })
    await wrapper.vm.$nextTick()
    expect(wrapper.vm.tableDate).toBe('2014')
  })

  it('should match snapshot with pick-month prop', () => {
    const wrapper = mountFunction({
      propsData: {
        value: '2013-05-07',
        type: 'month',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should match snapshot with allowed dates as array', () => {
    const wrapper = mountFunction({
      propsData: {
        value: '2013-05',
        type: 'month',
        allowedDates: value => ['2013-01', '2013-03', '2013-05', '2013-07'].includes(value),
      },
    })

    expect(wrapper.findAll('.v-date-picker-table--month tbody').at(0).html()).toMatchSnapshot()
  })

  it('should match snapshot with month formatting functions', () => {
    const wrapper = mountFunction({
      propsData: {
        value: '2005-11-01',
        type: 'month',
        monthFormat: date => `(${date.split('-')[1]})`,
      },
    })

    expect(wrapper.findAll('.v-date-picker-table--month tbody').at(0).html()).toMatchSnapshot()
  })

  it('should match snapshot with colored picker & header', () => {
    const wrapper = mountFunction({
      propsData: {
        type: 'month',
        value: '2005-11-01',
        color: 'primary',
        headerColor: 'orange darken-1',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should match snapshot with colored picker', () => {
    const wrapper = mountFunction({
      propsData: {
        type: 'month',
        value: '2005-11-01',
        color: 'orange darken-1',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should match change month when clicked on header arrow buttons', () => {
    const wrapper = mountFunction({
      propsData: {
        value: '2005-11',
        type: 'month',
      },
    })

    const [leftButton, rightButton] = wrapper.findAll('.v-date-picker-header button.v-btn').wrappers

    leftButton.trigger('click')
    expect(wrapper.vm.tableDate).toBe('2004')

    rightButton.trigger('click')
    expect(wrapper.vm.tableDate).toBe('2005')
  })

  it('should match change active picker when clicked on month button', () => {
    const wrapper = mountFunction({
      propsData: {
        value: '2005-11-01',
        type: 'month',
      },
    })

    const button = wrapper.findAll('.v-date-picker-header__value button').at(0)

    button.trigger('click')
    expect(wrapper.vm.internalActivePicker).toBe('YEAR')
  })

  it('should select year', async () => {
    const wrapper = mountFunction({
      propsData: {
        type: 'month',
        value: '2005-11',
      },
    })

    wrapper.setData({
      internalActivePicker: 'YEAR',
    })

    wrapper.findAll('.v-date-picker-years li.active + li').at(0).trigger('click')
    expect(wrapper.vm.internalActivePicker).toBe('MONTH')
    expect(wrapper.vm.tableDate).toBe('2004')
  })

  it('should set the table date when value has changed', () => {
    const wrapper = mountFunction({
      propsData: {
        value: null,
        type: 'month',
      },
    })

    wrapper.setProps({ value: '2005-11' })
    expect(wrapper.vm.tableDate).toBe('2005')
  })

  it('should use prev and next icons', () => {
    const wrapper = mountFunction({
      propsData: {
        type: 'month',
        prevIcon: 'block',
        nextIcon: 'check',
      },
    })

    const icons = wrapper.findAll('.v-date-picker-header .v-icon').wrappers
    expect(icons[0].element.textContent).toBe('block')
    expect(icons[1].element.textContent).toBe('check')
  })

  it('should display translated title', async () => {
    const wrapper = mountFunction({
      propsData: {
        multiple: true,
        type: 'month',
        value: ['2013-05'],
      },
    })

    expect(wrapper.find('.v-date-picker-title__date').text()).toBe('May')

    wrapper.setProps({
      value: [],
    })
    expect(wrapper.find('.v-date-picker-title__date').text()).toBe('-')

    wrapper.setProps({
      value: ['2013-05', '2013-06', '2013-07'],
    })
    expect(wrapper.find('.v-date-picker-title__date').text()).toBe('3 selected')
  })

  it('should emit click/dblclick:month event', async () => {
    const click = jest.fn()
    const dblclick = jest.fn()
    const wrapper = mountFunction({
      propsData: {
        value: '2013-05',
        type: 'month',
      },
      listeners: {
        'click:month': (value: any, event: any) => click(value, event instanceof Event),
        'dblclick:month': (value: any, event: any) => dblclick(value, event instanceof Event),
      },
    })

    wrapper.findAll('.v-date-picker-table--month tbody tr+tr td:first-child button').at(0).trigger('click')
    expect(click).toHaveBeenCalledWith('2013-04', true)

    wrapper.findAll('.v-date-picker-table--month tbody tr+tr td:first-child button').at(0).trigger('dblclick')
    expect(dblclick).toHaveBeenCalledWith('2013-04', true)
  })

  it('should handle date range select', async () => {
    const cb = jest.fn()
    const wrapper = mountFunction({
      propsData: {
        range: true,
        type: 'month',
        value: [],
      },
    })
    const year = new Date().getFullYear()
    const toDate = `${year}-08`
    const fromDate = `${year}-03`

    wrapper.vm.$on('input', cb)
    wrapper.find('.v-date-picker-table--month tbody tr:first-child td:nth-child(3) button').trigger('click')
    expect(cb.mock.calls[0][0]).toEqual(
      expect.arrayContaining([fromDate])
    )

    wrapper.find('.v-date-picker-table--month tbody tr:first-child+tr+tr td:nth-child(2) button').trigger('click')
    expect(cb.mock.calls[0][0][0]).toBe(fromDate)
    expect(cb.mock.calls[1][0][0]).toBe(toDate)
  })

  it('should add class for the first and last days in range', async () => {
    const wrapper = mountFunction({
      propsData: {
        range: true,
        showCurrent: '2019',
        type: 'month',
        value: ['2019-01', '2019-02'],
      },
    })

    expect(wrapper.findAll('.v-date-picker-table--month tbody button.v-date-picker--first-in-range')
      .exists()).toBe(true)
    expect(wrapper.findAll('.v-date-picker-table--month tbody button.v-date-picker--last-in-range')
      .exists()).toBe(true)
  })
})
