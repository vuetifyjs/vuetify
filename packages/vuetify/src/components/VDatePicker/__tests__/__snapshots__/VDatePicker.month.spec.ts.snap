// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VDatePicker.ts should match snapshot with allowed dates as array 1`] = `
<tbody>
  <tr>
    <td>
      <button type="button"
              class="v-btn v-size--default v-btn--text theme--light"
      >
        <div class="v-btn__content">
          Jan
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              class="v-btn v-size--default v-btn--flat v-btn--text v-btn--disabled theme--light"
              disabled
      >
        <div class="v-btn__content">
          Feb
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              class="v-btn v-size--default v-btn--text theme--light"
      >
        <div class="v-btn__content">
          Mar
        </div>
      </button>
    </td>
  </tr>
  <tr>
    <td>
      <button type="button"
              class="v-btn v-size--default v-btn--flat v-btn--text v-btn--disabled theme--light"
              disabled
      >
        <div class="v-btn__content">
          Apr
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              class="v-btn v-size--default v-btn--active theme--light accent"
      >
        <div class="v-btn__content">
          May
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              class="v-btn v-size--default v-btn--flat v-btn--text v-btn--disabled theme--light"
              disabled
      >
        <div class="v-btn__content">
          Jun
        </div>
      </button>
    </td>
  </tr>
  <tr>
    <td>
      <button type="button"
              class="v-btn v-size--default v-btn--text theme--light"
      >
        <div class="v-btn__content">
          Jul
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              class="v-btn v-size--default v-btn--flat v-btn--text v-btn--disabled theme--light"
              disabled
      >
        <div class="v-btn__content">
          Aug
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              class="v-btn v-size--default v-btn--flat v-btn--text v-btn--disabled theme--light"
              disabled
      >
        <div class="v-btn__content">
          Sep
        </div>
      </button>
    </td>
  </tr>
  <tr>
    <td>
      <button type="button"
              class="v-btn v-size--default v-btn--flat v-btn--text v-btn--disabled theme--light"
              disabled
      >
        <div class="v-btn__content">
          Oct
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              class="v-btn v-size--default v-btn--flat v-btn--text v-btn--disabled theme--light"
              disabled
      >
        <div class="v-btn__content">
          Nov
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              class="v-btn v-size--default v-btn--flat v-btn--text v-btn--disabled theme--light"
              disabled
      >
        <div class="v-btn__content">
          Dec
        </div>
      </button>
    </td>
  </tr>
</tbody>
`;

exports[`VDatePicker.ts should match snapshot with colored picker & header 1`] = `
<div class="v-picker v-card v-picker--date theme--light">
  <div class="v-picker__title orange darken-1">
    <div class="v-date-picker-title">
      <div class="v-picker__title__btn v-date-picker-title__year">
        2005
      </div>
      <div class="v-picker__title__btn v-date-picker-title__date v-picker__title__btn--active">
        <div>
          November
        </div>
      </div>
    </div>
  </div>
  <div class="v-picker__body theme--light"
       style="width: 290px;"
  >
    <div>
      <div class="v-date-picker-header theme--light">
        <button type="button"
                class="v-btn v-btn--icon v-btn--round theme--light v-size--default"
                aria-label="Previous year"
        >
          <span class="v-btn__content">
            <i aria-hidden="true"
               class="v-icon notranslate mdi mdi-chevron-left theme--light"
            >
            </i>
          </span>
        </button>
        <div class="v-date-picker-header__value">
          <div class="primary--text">
            <button type="button">
              2005
            </button>
          </div>
        </div>
        <button type="button"
                class="v-btn v-btn--icon v-btn--round theme--light v-size--default"
                aria-label="Next year"
        >
          <span class="v-btn__content">
            <i aria-hidden="true"
               class="v-icon notranslate mdi mdi-chevron-right theme--light"
            >
            </i>
          </span>
        </button>
      </div>
      <div class="v-date-picker-table v-date-picker-table--month theme--light">
        <table>
          <tbody>
            <tr>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Jan
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Feb
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Mar
                  </div>
                </button>
              </td>
            </tr>
            <tr>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Apr
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    May
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Jun
                  </div>
                </button>
              </td>
            </tr>
            <tr>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Jul
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Aug
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Sep
                  </div>
                </button>
              </td>
            </tr>
            <tr>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Oct
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Nov
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Dec
                  </div>
                </button>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;

exports[`VDatePicker.ts should match snapshot with colored picker 1`] = `
<div class="v-picker v-card v-picker--date theme--light">
  <div class="v-picker__title orange darken-1">
    <div class="v-date-picker-title">
      <div class="v-picker__title__btn v-date-picker-title__year">
        2005
      </div>
      <div class="v-picker__title__btn v-date-picker-title__date v-picker__title__btn--active">
        <div>
          November
        </div>
      </div>
    </div>
  </div>
  <div class="v-picker__body theme--light"
       style="width: 290px;"
  >
    <div>
      <div class="v-date-picker-header theme--light">
        <button type="button"
                class="v-btn v-btn--icon v-btn--round theme--light v-size--default"
                aria-label="Previous year"
        >
          <span class="v-btn__content">
            <i aria-hidden="true"
               class="v-icon notranslate mdi mdi-chevron-left theme--light"
            >
            </i>
          </span>
        </button>
        <div class="v-date-picker-header__value">
          <div class="orange--text text--darken-1">
            <button type="button">
              2005
            </button>
          </div>
        </div>
        <button type="button"
                class="v-btn v-btn--icon v-btn--round theme--light v-size--default"
                aria-label="Next year"
        >
          <span class="v-btn__content">
            <i aria-hidden="true"
               class="v-icon notranslate mdi mdi-chevron-right theme--light"
            >
            </i>
          </span>
        </button>
      </div>
      <div class="v-date-picker-table v-date-picker-table--month theme--light">
        <table>
          <tbody>
            <tr>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Jan
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Feb
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Mar
                  </div>
                </button>
              </td>
            </tr>
            <tr>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Apr
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    May
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Jun
                  </div>
                </button>
              </td>
            </tr>
            <tr>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Jul
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Aug
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Sep
                  </div>
                </button>
              </td>
            </tr>
            <tr>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Oct
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Nov
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Dec
                  </div>
                </button>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;

exports[`VDatePicker.ts should match snapshot with month formatting functions 1`] = `
<tbody>
  <tr>
    <td>
      <button type="button"
              class="v-btn v-size--default v-btn--text theme--light"
      >
        <div class="v-btn__content">
          (01)
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              class="v-btn v-size--default v-btn--text theme--light"
      >
        <div class="v-btn__content">
          (02)
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              class="v-btn v-size--default v-btn--text theme--light"
      >
        <div class="v-btn__content">
          (03)
        </div>
      </button>
    </td>
  </tr>
  <tr>
    <td>
      <button type="button"
              class="v-btn v-size--default v-btn--text theme--light"
      >
        <div class="v-btn__content">
          (04)
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              class="v-btn v-size--default v-btn--text theme--light"
      >
        <div class="v-btn__content">
          (05)
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              class="v-btn v-size--default v-btn--text theme--light"
      >
        <div class="v-btn__content">
          (06)
        </div>
      </button>
    </td>
  </tr>
  <tr>
    <td>
      <button type="button"
              class="v-btn v-size--default v-btn--text theme--light"
      >
        <div class="v-btn__content">
          (07)
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              class="v-btn v-size--default v-btn--text theme--light"
      >
        <div class="v-btn__content">
          (08)
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              class="v-btn v-size--default v-btn--text theme--light"
      >
        <div class="v-btn__content">
          (09)
        </div>
      </button>
    </td>
  </tr>
  <tr>
    <td>
      <button type="button"
              class="v-btn v-size--default v-btn--text theme--light"
      >
        <div class="v-btn__content">
          (10)
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              class="v-btn v-size--default v-btn--text theme--light"
      >
        <div class="v-btn__content">
          (11)
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              class="v-btn v-size--default v-btn--text theme--light"
      >
        <div class="v-btn__content">
          (12)
        </div>
      </button>
    </td>
  </tr>
</tbody>
`;

exports[`VDatePicker.ts should match snapshot with pick-month prop 1`] = `
<div class="v-picker v-card v-picker--date theme--light">
  <div class="v-picker__title primary">
    <div class="v-date-picker-title">
      <div class="v-picker__title__btn v-date-picker-title__year">
        2013
      </div>
      <div class="v-picker__title__btn v-date-picker-title__date v-picker__title__btn--active">
        <div>
          May
        </div>
      </div>
    </div>
  </div>
  <div class="v-picker__body theme--light"
       style="width: 290px;"
  >
    <div>
      <div class="v-date-picker-header theme--light">
        <button type="button"
                class="v-btn v-btn--icon v-btn--round theme--light v-size--default"
                aria-label="Previous year"
        >
          <span class="v-btn__content">
            <i aria-hidden="true"
               class="v-icon notranslate mdi mdi-chevron-left theme--light"
            >
            </i>
          </span>
        </button>
        <div class="v-date-picker-header__value">
          <div class="accent--text">
            <button type="button">
              2013
            </button>
          </div>
        </div>
        <button type="button"
                class="v-btn v-btn--icon v-btn--round theme--light v-size--default"
                aria-label="Next year"
        >
          <span class="v-btn__content">
            <i aria-hidden="true"
               class="v-icon notranslate mdi mdi-chevron-right theme--light"
            >
            </i>
          </span>
        </button>
      </div>
      <div class="v-date-picker-table v-date-picker-table--month theme--light">
        <table>
          <tbody>
            <tr>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Jan
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Feb
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Mar
                  </div>
                </button>
              </td>
            </tr>
            <tr>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Apr
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    May
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Jun
                  </div>
                </button>
              </td>
            </tr>
            <tr>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Jul
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Aug
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Sep
                  </div>
                </button>
              </td>
            </tr>
            <tr>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Oct
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Nov
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Dec
                  </div>
                </button>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;

exports[`VDatePicker.ts should render flat picker 1`] = `
<div class="v-picker v-card v-picker--date v-picker--flat theme--light">
  <div class="v-picker__title primary">
    <div class="v-date-picker-title">
      <div class="v-picker__title__btn v-date-picker-title__year">
        2013
      </div>
      <div class="v-picker__title__btn v-date-picker-title__date v-picker__title__btn--active">
        <div>
          May
        </div>
      </div>
    </div>
  </div>
  <div class="v-picker__body theme--light"
       style="width: 290px;"
  >
    <div>
      <div class="v-date-picker-header theme--light">
        <button type="button"
                class="v-btn v-btn--icon v-btn--round theme--light v-size--default"
                aria-label="Previous year"
        >
          <span class="v-btn__content">
            <i aria-hidden="true"
               class="v-icon notranslate mdi mdi-chevron-left theme--light"
            >
            </i>
          </span>
        </button>
        <div class="v-date-picker-header__value">
          <div class="accent--text">
            <button type="button">
              2013
            </button>
          </div>
        </div>
        <button type="button"
                class="v-btn v-btn--icon v-btn--round theme--light v-size--default"
                aria-label="Next year"
        >
          <span class="v-btn__content">
            <i aria-hidden="true"
               class="v-icon notranslate mdi mdi-chevron-right theme--light"
            >
            </i>
          </span>
        </button>
      </div>
      <div class="v-date-picker-table v-date-picker-table--month theme--light">
        <table>
          <tbody>
            <tr>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Jan
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Feb
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Mar
                  </div>
                </button>
              </td>
            </tr>
            <tr>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Apr
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--active theme--light accent"
                >
                  <div class="v-btn__content">
                    May
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Jun
                  </div>
                </button>
              </td>
            </tr>
            <tr>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Jul
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Aug
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Sep
                  </div>
                </button>
              </td>
            </tr>
            <tr>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Oct
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Nov
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Dec
                  </div>
                </button>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;

exports[`VDatePicker.ts should render picker with elevation 1`] = `
<div class="v-picker v-card v-picker--date theme--light elevation-15">
  <div class="v-picker__title primary">
    <div class="v-date-picker-title">
      <div class="v-picker__title__btn v-date-picker-title__year">
        2013
      </div>
      <div class="v-picker__title__btn v-date-picker-title__date v-picker__title__btn--active">
        <div>
          May
        </div>
      </div>
    </div>
  </div>
  <div class="v-picker__body theme--light"
       style="width: 290px;"
  >
    <div>
      <div class="v-date-picker-header theme--light">
        <button type="button"
                class="v-btn v-btn--icon v-btn--round theme--light v-size--default"
                aria-label="Previous year"
        >
          <span class="v-btn__content">
            <i aria-hidden="true"
               class="v-icon notranslate mdi mdi-chevron-left theme--light"
            >
            </i>
          </span>
        </button>
        <div class="v-date-picker-header__value">
          <div class="accent--text">
            <button type="button">
              2013
            </button>
          </div>
        </div>
        <button type="button"
                class="v-btn v-btn--icon v-btn--round theme--light v-size--default"
                aria-label="Next year"
        >
          <span class="v-btn__content">
            <i aria-hidden="true"
               class="v-icon notranslate mdi mdi-chevron-right theme--light"
            >
            </i>
          </span>
        </button>
      </div>
      <div class="v-date-picker-table v-date-picker-table--month theme--light">
        <table>
          <tbody>
            <tr>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Jan
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Feb
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Mar
                  </div>
                </button>
              </td>
            </tr>
            <tr>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Apr
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--active theme--light accent"
                >
                  <div class="v-btn__content">
                    May
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Jun
                  </div>
                </button>
              </td>
            </tr>
            <tr>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Jul
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Aug
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Sep
                  </div>
                </button>
              </td>
            </tr>
            <tr>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Oct
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Nov
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="v-btn v-size--default v-btn--text theme--light"
                >
                  <div class="v-btn__content">
                    Dec
                  </div>
                </button>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
